### PR TITLE
Update DevWorkspace samples to be current

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
Update the `samples/` repo to remove out-of-date samples (e.g. old flattened theia-next workspaces that no longer work) and add current samples (e.g. the DevWorkspace used by the Web Terminal Operator). 

Currently there is no sample for Code-OSS, as that depends on https://github.com/devfile/devworkspace-operator/pull/844

### What issues does this PR fix or reference?
Out of date samples were confusing (flattened samples do not work anymore since they haven't been updated)

### Is it tested? How?
All samples should give a functioning workspace (web terminal samples require an OpenShift cluster)

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v8-devworkspace-operator-e2e, v8-che-happy-path` to trigger)
    - [ ] `v8-devworkspace-operator-e2e`: DevWorkspace e2e test
    - [ ] `v8-che-happy-path`: Happy path for verification integration with Che

### CODE REVIEW

File: samples/git-clone-sample [hunk 1]
```diff
@@ -0,0 +1,36 @@
+kind: DevWorkspace
```
Comment (AObuchow) [inline samples/git-clone-sample:1]: This file is missing the `.yaml` extension.

Comment (amisevsk) [reply samples/git-clone-sample:1]: Woopsie :)

**Review by AObuchow — APPROVED** [review]
Looks good to me, this update was well needed :)

**Comment by dkwon17** [comment]
I don't know if I did something wrong (can test again tomorrow) but I wasn't able to get the `samples/web-terminal-flattened.yaml` and `samples/web-terminal.yaml` workspaces to start; they're stuck at `Starting` for me:

After running:
```
oc apply -f samples/web-terminal-flattened.yaml
```

I get the following:

The devworkspace that's stuck at `Starting`:
image

The workspace pod:
image

The `web-terminal-exec` container log (there were no logs shown for `web-terminal-tooling`):
image

**Comment by amisevsk** [comment]
@dkwon17 Hm, interesting. The status of your DevWorkspace suggests that DWO is "waiting for editor to start", meaning it's not getting a 2xx or 4xx response on the mainUrl for the workspace. 

Are there any logs in DWO that look out of place? Are there any networkPolicies in your cluster that prevent communication between the DWO namespace and the terminal's namespace?

**Review by ibuziuk — APPROVED** [review]

**Review by dkwon17 — APPROVED** [review]
I was able to get the webterminal workspaces running when trying on a cluster with no added network policies in my user namespace

**Pull request merged**